src/bridge.cpp

```diff
@@ -1,24 +1,133 @@
     Polygons islands;
+
+    Polygons prevLayerOutline; // we also want the complete outline of the previous layer
+
     for(auto prevLayerPart : prevLayer->parts)
     {
+        prevLayerOutline.add(prevLayerPart.outline); // not intersected with skin
+
         if (!boundaryBox.hit(prevLayerPart.boundaryBox))
             continue;
         
-        islands.add(outline.intersection(prevLayerPart.outline));
+        islands.add(skinOutline.intersection(prevLayerPart.outline));
     }
     supportedRegions = islands;
+
+    if (supportLayer)
+    {
+        // add the regions of the skin that have support below them to supportedRegions
+        // but don't add these regions to islands because that can actually cause the code
+        // below to consider the skin a bridge when it isn't (e.g. a skin that is supported by
+        // the model on one side but the remainder of the skin is above support would look like
+        // a bridge because it would have two islands) - FIXME more work required here?
+
+        if (!supportLayer->support_roof.empty())
+        {
+            AABB support_roof_bb(supportLayer->support_roof);
+            if (boundaryBox.hit(support_roof_bb))
+            {
+                prevLayerOutline.add(supportLayer->support_roof); // not intersected with skin
+
+                Polygons supported_skin(skinOutline.intersection(supportLayer->support_roof));
+                if (!supported_skin.empty())
+                {
+                    supportedRegions.add(supported_skin);
+                }
+            }
+        }
+        else
+        {
+            for (auto support_part : supportLayer->support_infill_parts)
+            {
+                AABB support_part_bb(support_part.getInfillArea());
+                if (boundaryBox.hit(support_part_bb))
+                {
+                    prevLayerOutline.add(support_part.getInfillArea()); // not intersected with skin
+
+                    Polygons supported_skin(skinOutline.intersection(support_part.getInfillArea()));
+                    if (!supported_skin.empty())
+                    {
+                        supportedRegions.add(supported_skin);
+                    }
+                }
+            }
+        }
+    }
+
+    // if the proportion of the skin region that is supported is less than supportThreshold, it's considered a bridge and we
+    // determine the best angle for the skin lines - the current heuristic is that the skin lines should be parallel to the
+    // direction of the skin area's longest unsupported edge - if the skin has no unsupported edges, we fall through to the
+    // original code
+
+    if (supportThreshold > 0 && (supportedRegions.area() / (skinOutline.area() + 1)) < supportThreshold)
+    {
+        Polygons bbPoly;
+        bbPoly.add(boundaryBox.toPolygon());
+
+        // airBelow is the region below the skin that is not supported, it extends well past the boundary of the skin.
```

Comment:
I can image that in most cases this would work fine. However, sometimes you do want the old method I guess? Like in this hypothetical example.
![bridging-direction](https://user-images.githubusercontent.com/1016604/40483263-b29af800-5f57-11e8-8096-c9b76ea7648c.png)
[wheel-cylinder-thingy2.stl.zip](https://github.com/Ultimaker/CuraEngine/files/2035106/wheel-cylinder-thingy2.stl.zip)

Reply:
Except that is impossible to print whatever the direction of the skin lines!